File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    complete = []
    for i in range(len(progresses)):
        day = math.ceil((100 - progresses[i]) / speeds[i])
        complete.append(day)
    
    count = 1
    i = 0
    for j in range(i + 1, len(complete)):
        if complete[i] >= complete[j]:
            count += 1
        else:
            i = j
            answer.append(count)
            count = 1
    answer.append(count)

    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();

        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < progresses.length; i++) {
            list.add((int) Math.ceil((double) (100 - progresses[i]) / speeds[i]));
        }

        int pre = list.get(0);
        int count = 1;
        for (int i = 1; i < list.size(); i++) {
            if (list.get(i) <= pre) {
                count++;
            } else {
                answer.add(count);
                pre = list.get(i);
                count = 1;
            }
        }
        answer.add(count);

        return answer.stream().mapToInt(i -> i).toArray();
    }
}